src/infrastructure/jobs/AbstractRedisJobProcessor.ts

```diff
@@ -0,0 +1,39 @@
+import type { BackgroundJobProcessorConfig, BaseJobPayload } from '@lokalise/background-jobs-common'
+import {
+  AbstractBackgroundJobProcessor,
+  CommonBullmqFactory,
+} from '@lokalise/background-jobs-common'
+import type { RequestContext } from '@lokalise/fastify-extras'
+import type { Job } from 'bullmq'
+
+import { isTest } from '../config'
+import type { Dependencies } from '../diConfig'
+
+export type BackgroundJobConfig = Omit<BackgroundJobProcessorConfig, 'isTest'>
+
+export abstract class AbstractRedisJobProcessor<
```

Comment:
If we decide to rename the `InMemory` job, I guess we can just rename this as `AbstractJobProcessor` or something like that but no strong opinion here

Reply:
yeah, you are rigjht ... 😓 not sure which is a good pattern here sorry, if yo think that keeping Redis in the name is fine, I am ok with that

Reply:
```AbstractEnqueuedJobProcessor```? Not the best with naming today:)

Reply:
I'm scared of what you can do when you are at your best then, mortals are not supposed to wield that much power.

This is perfect, thanks!